### DISCUSSION TITLE
Uris array not updating

### DISCUSSION THREAD

**Discussion by Fenaz12** [gilbarbara/react-spotify-web-playback#99] [Q&A] [answered]
I am trying to add a state of array as uris but its only works when it runs for the first time and not getting updated inside the SpotifyPlayer, The state is updating perfectly but It doesn't get passed to the player whenever it is updated. Is it possible to do that? 

Edit: It works perfectly when I use a single uri, It is also a state which changes on click but, it doesn't work when I use array of uris instead of single uri.

**Comment by gilbarbara** [comment, marked as answer]
Hey @Fenaz12 
I'm using it that way in the demo. Check the [source](https://github.com/gilbarbara/react-spotify-web-playback/blob/main/demo/src/App.tsx)

**Reply by Fenaz12** [reply]
Hi there,
From what I saw in that code, you already have specified the URIs in the array If I am correct, Please correct me If I am wrong. But I am trying to push the URI to an already existing array whenever I click a certain URI, So the player will show the next and previous buttons which don't happen when I use a single URI instead of an array. The problem is whenever I tried to add the URI to the array, It doesn't get updated inside the player except for the first time.
And also thank you very much for the API it really makes my life and project easy and better :)

**Reply by gilbarbara** [reply]
Are you mutating the `uris` directly? This wouldn't trigger an update.
Create a `uris` state and update it when you add/remove something so the Player knows it changed.
